pkg/execution/driver/driver.go

```diff
@@ -64,8 +73,34 @@ func MarshalV1(
 
 	j, err := json.Marshal(req)
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("error marshalling request to JSON: %w", err)
+	}
+
+	b, err := jcs.Transform(j)
+	if err != nil {
+		return nil, fmt.Errorf("error transforming request with JCS: %w", err)
 	}
 
-	return jcs.Transform(j)
+	return b, nil
+}
+
+// mapInvocationEventNames sets the name of any invocation events to the name of the trigger event.
+// This is used to ensure that SDKs receive the event they are expecting when invoked.
+func mapInvocationEventNames(events []map[string]any, fn inngest.Function) ([]map[string]any, error) {
+	for _, evt := range events {
+		if name, ok := evt["name"].(string); !ok || name != event.InvokeFnName {
+			continue
+		}
+
+		if len(fn.Triggers) != 1 {
+			return nil, fmt.Errorf("invocation event found, but function %s has %d triggers; could not fill in invocation event name automatically", fn.Slug, len(fn.Triggers))
+		}
+
+		trigger := fn.Triggers[0]
+		// Crons can keep the original event name
+		if trigger.EventTrigger != nil && trigger.Event != "" {
+			evt["name"] = trigger.Event
```

Comment:
So, I get it but I'm wondering why this is necessary for the SDK?

A few things that could go wrong:

- It might make external logging lie.  We didn't actually receive this event name.  On the user's side, they'll see many more events of this type than they're actually sending.
- The events received from the SDK won't correlate with our dashboard

Reply:
@tonyhb Two paths here: we can do this or make `event.name` in the SDK be a union of `"the-actual-event" | "inngest/function.invoked"`.

I don't have a strong opinion on this either way - the latter might feel a touch confusing if a user doesn't know what invocation is, but it also provides a discovery path.

I'll delete this code and the SDK can declare it.